File: tutorials/SampleCompany/Simple/SampleServer/Model/Constants/Python/samplecompanysimpleservermodel_constants.py
from enum import Enum

class Namespaces(Enum):
     Uri = "http://samplecompany.com/SimpleServer/Model"

class BrowseNames(Enum):
    CurrentStep = "CurrentStep"
    CycleId = "CycleId"
    CycleStepDataType = "CycleStepDataType"
    Error = "Error"
    SimpleServer_BinarySchema = "SampleCompany.SimpleServer.Model"
    SimpleServer_XmlSchema = "SampleCompany.SimpleServer.Model"
    Steps = "Steps"
    SystemCycleAbortedEventType = "SystemCycleAbortedEventType"
    SystemCycleFinishedEventType = "SystemCycleFinishedEventType"
    SystemCycleStartedEventType = "SystemCycleStartedEventType"
    SystemCycleStatusEventType = "SystemCycleStatusEventType"

class DataTypeIds(Enum):
    CycleStepDataType = "nsu=http://samplecompany.com/SimpleServer/Model;i=1"

def get_DataTypeIds_name(value: str) -> str:
    try:
        return DataTypeIds(value).name
    except ValueError:
        return None


class ObjectIds(Enum):
    CycleStepDataType_Encoding_DefaultBinary = "nsu=http://samplecompany.com/SimpleServer/Model;i=52"
    CycleStepDataType_Encoding_DefaultXml = "nsu=http://samplecompany.com/SimpleServer/Model;i=60"
    CycleStepDataType_Encoding_DefaultJson = "nsu=http://samplecompany.com/SimpleServer/Model;i=68"

def get_ObjectIds_name(value: str) -> str:
    try:
        return ObjectIds(value).name
    except ValueError:
        return None


class ObjectTypeIds(Enum):
    SystemCycleStatusEventType = "nsu=http://samplecompany.com/SimpleServer/Model;i=2"
    SystemCycleStartedEventType = "nsu=http://samplecompany.com/SimpleServer/Model;i=14"
    SystemCycleAbortedEventType = "nsu=http://samplecompany.com/SimpleServer/Model;i=27"
    SystemCycleFinishedEventType = "nsu=http://samplecompany.com/SimpleServer/Model;i=40"

def get_ObjectTypeIds_name(value: str) -> str:
    try:
        return ObjectTypeIds(value).name
    except ValueError:
        return None


class VariableIds(Enum):
    SystemCycleStatusEventType_CycleId = "nsu=http://samplecompany.com/SimpleServer/Model;i=12"
    SystemCycleStatusEventType_CurrentStep = "nsu=http://samplecompany.com/SimpleServer/Model;i=13"
    SystemCycleStartedEventType_Steps = "nsu=http://samplecompany.com/SimpleServer/Model;i=26"
    SystemCycleAbortedEventType_Error = "nsu=http://samplecompany.com/SimpleServer/Model;i=39"
    SimpleServer_BinarySchema = "nsu=http://samplecompany.com/SimpleServer/Model;i=53"
    SimpleServer_BinarySchema_NamespaceUri = "nsu=http://samplecompany.com/SimpleServer/Model;i=55"
    SimpleServer_BinarySchema_Deprecated = "nsu=http://samplecompany.com/SimpleServer/Model;i=56"
    SimpleServer_BinarySchema_CycleStepDataType = "nsu=http://samplecompany.com/SimpleServer/Model;i=57"
    SimpleServer_XmlSchema = "nsu=http://samplecompany.com/SimpleServer/Model;i=61"
    SimpleServer_XmlSchema_NamespaceUri = "nsu=http://samplecompany.com/SimpleServer/Model;i=63"
    SimpleServer_XmlSchema_Deprecated = "nsu=http://samplecompany.com/SimpleServer/Model;i=64"
    SimpleServer_XmlSchema_CycleStepDataType = "nsu=http://samplecompany.com/SimpleServer/Model;i=65"

def get_VariableIds_name(value: str) -> str:
    try:
        return VariableIds(value).name
    except ValueError:
        return None

File: tutorials/SampleCompany/Simple/SampleServer/Model/Constants/TypeScript/samplecompanysimpleservermodel_constants.ts
export const NS = 'http://samplecompany.com/SimpleServer/Model';

export class BrowseNames {
   static readonly CurrentStep: string = 'CurrentStep'
   static readonly CycleId: string = 'CycleId'
   static readonly CycleStepDataType: string = 'CycleStepDataType'
   static readonly Error: string = 'Error'
   static readonly SimpleServer_BinarySchema: string = 'SampleCompany.SimpleServer.Model'
   static readonly SimpleServer_XmlSchema: string = 'SampleCompany.SimpleServer.Model'
   static readonly Steps: string = 'Steps'
   static readonly SystemCycleAbortedEventType: string = 'SystemCycleAbortedEventType'
   static readonly SystemCycleFinishedEventType: string = 'SystemCycleFinishedEventType'
   static readonly SystemCycleStartedEventType: string = 'SystemCycleStartedEventType'
   static readonly SystemCycleStatusEventType: string = 'SystemCycleStatusEventType'
}

export class DataTypeIds {
    static readonly CycleStepDataType: string = 'nsu=' + NS + ';i=1'
}

export class ObjectIds {
    static readonly CycleStepDataType_Encoding_DefaultBinary: string = 'nsu=' + NS + ';i=52'
    static readonly CycleStepDataType_Encoding_DefaultXml: string = 'nsu=' + NS + ';i=60'
    static readonly CycleStepDataType_Encoding_DefaultJson: string = 'nsu=' + NS + ';i=68'
}

export class ObjectTypeIds {
    static readonly SystemCycleStatusEventType: string = 'nsu=' + NS + ';i=2'
    static readonly SystemCycleStartedEventType: string = 'nsu=' + NS + ';i=14'
    static readonly SystemCycleAbortedEventType: string = 'nsu=' + NS + ';i=27'
    static readonly SystemCycleFinishedEventType: string = 'nsu=' + NS + ';i=40'
}

export class VariableIds {
    static readonly SystemCycleStatusEventType_CycleId: string = 'nsu=' + NS + ';i=12'
    static readonly SystemCycleStatusEventType_CurrentStep: string = 'nsu=' + NS + ';i=13'
    static readonly SystemCycleStartedEventType_Steps: string = 'nsu=' + NS + ';i=26'
    static readonly SystemCycleAbortedEventType_Error: string = 'nsu=' + NS + ';i=39'
    static readonly SimpleServer_BinarySchema: string = 'nsu=' + NS + ';i=53'
    static readonly SimpleServer_BinarySchema_NamespaceUri: string = 'nsu=' + NS + ';i=55'
    static readonly SimpleServer_BinarySchema_Deprecated: string = 'nsu=' + NS + ';i=56'
    static readonly SimpleServer_BinarySchema_CycleStepDataType: string = 'nsu=' + NS + ';i=57'
    static readonly SimpleServer_XmlSchema: string = 'nsu=' + NS + ';i=61'
    static readonly SimpleServer_XmlSchema_NamespaceUri: string = 'nsu=' + NS + ';i=63'
    static readonly SimpleServer_XmlSchema_Deprecated: string = 'nsu=' + NS + ';i=64'
    static readonly SimpleServer_XmlSchema_CycleStepDataType: string = 'nsu=' + NS + ';i=65'
}
